src/game.rs

```diff
@@ -0,0 +1,250 @@
+        for dx in -1..=1 {
+            let nx = x + dx;
+            if nx < 0 || nx >= self.width as i32 {
+                // checks if neighbor's x is out of bounds
+                continue;
+            }
+            for dy in -1..=1 {
+                let ny = y + dy;
+                if ny < 0 || ny >= self.height as i32 {
+                    // checks if neighbor's y is out of bounds
+                    continue;
+                }
+                if nx == x && ny == y {
+                    continue;
+                }
+                let index = self.get_index(nx, ny);
+                if self.cells[index] {
+                    count += 1;
+                }
+            }
+        }
+        count
+    }
+
+    /// Updates the game state to the next generation.
+    fn update(&mut self) {
+        let mut next_cells = vec![false; self.height * self.width];
+        for x in 0..self.width as i32 {
+            for y in 0..self.height as i32 {
+                let index = self.get_index(x, y);
+                let cell = self.cells[index];
+                let neighbors = self.count_neighbors(x, y);
+
+                next_cells[index] = matches!((cell, neighbors), (true, 2) | (true, 3) | (false, 3));
+            }
+        }
+        self.cells = next_cells
+    }
+
+    /// Draws the current game state.
+    ///
+    /// # Arguments
+    ///
+    /// * `cell_size` - The size of each cell.
+    fn draw(&self, cell_size: f32) {
+        for x in 0..self.width {
+            for y in 0..self.height {
+                let index = y * self.width + x;
+                if self.cells[index] {
+                    draw_rectangle(
+                        x as f32 * cell_size,
+                        y as f32 * cell_size,
+                        cell_size,
+                        cell_size,
+                        WHITE,
+                    );
+                }
+            }
+        }
+    }
+}
+
+/// Validates the start parameters for the game.
+///
+/// # Arguments
+///
+/// * `height` - The height of the game grid.
+/// * `width` - The width of the game grid.
+/// * `cell_size` - The size of each cell.
+///
+/// # Returns
+///
+/// `Ok(())` if the parameters are valid, otherwise an error message.
+fn validate_start_parameters(height: usize, width: usize, cell_size: f32) -> Result<(), String> {
+    if height == 0 || width == 0 {
+        return Err("Neither height nor width can be 0".to_string());
+    }
+    if cell_size <= 0.0 {
+        return Err("Cell size can't be negative or 0".to_string());
+    }
+    Ok(())
+}
+
+/// Starts the Game of Life.
+///
+/// # Arguments
+///
+/// * `height` - The height of the game grid.
+/// * `width` - The width of the game grid.
+/// * `cell_size` - The size of each cell.
+///
+/// # Returns
+///
+/// A message if an error ocurrs while validating the parameters, otherwise it loops infinitely.
+pub async fn start(height: usize, width: usize, cell_size: f32) -> Result<(), String> {
```

Comment:
I would put the game loop inside the `Game` struct, since they are closely related and, indeed, it is the whole point of the game.